tests/acceptance/dex/dex_test.go

```diff
@@ -1,122 +1,71 @@
+func TestSpec(t *testing.T) {
+	cfg := Config{}
+	err := envconfig.Init(&cfg)
 	if err != nil {
-		glog.Warningf("Unable to resolve host '%s' (if you are running this test from outside of Kyma ignore this log). Root cause: %v", ingressGatewayControllerServiceURL, err)
-
-		minikubeIp := tryToGetMinikubeIp()
-		if minikubeIp == "" {
-			t.Fatal(err)
-		}
-		ingressGatewayControllerAddr = minikubeIp
-	} else {
-		ingressGatewayControllerAddr = hostLookupResponse[0]
+		t.Errorf("Error while loading env config: %s", err)
 	}
 
-	domain, envFound := os.LookupEnv(domainEnvName)
-	if !envFound {
-		t.Fatal(domainEnvName + " env variable not set")
+	if !cfg.IsLocalEnv {
+		t.Skip("Test is enabled only in local env")
 	}
 
-	username, envFound := os.LookupEnv(usernameEnvName)
-	if !envFound {
-		t.Fatal(usernameEnvName + " env variable not set")
-	}
-
-	password, envFound := os.LookupEnv(passwordEnvName)
-	if !envFound {
-		t.Fatal(passwordEnvName + " env variable not set")
-	}
-
-	tr := &http.Transport{
-		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+	ingressClient, err := ingressgateway.FromEnv().Client()
+	if err != nil {
+		t.Errorf("Error while creating ingress gateway client: %s", err)
 	}
 
-	if testRunningInLocalEnv {
-		tr.DialContext = func(ctx context.Context, network, addr string) (net.Conn, error) {
-			// Changes request destination address to ingress gateway internal cluster address for requests to dex.
-			if strings.HasPrefix(addr, "dex") {
-				addr = fmt.Sprintf("%s:443", ingressGatewayControllerAddr)
-			}
-			dialer := net.Dialer{}
-			return dialer.DialContext(ctx, network, addr)
-		}
+	ingressClient.CheckRedirect = func(req *http.Request, via []*http.Request) error {
+		return http.ErrUseLastResponse
 	}
 
-	dexHttpClient := &http.Client{
-		Transport: tr,
-		Timeout:   time.Second * 10,
-		CheckRedirect: func(req *http.Request, via []*http.Request) error {
-			return http.ErrUseLastResponse
-		},
-	}
+	client := resilient.WrapHttpClient(ingressClient)
 
 	idProviderConfig := idProviderConfig{
 		dexConfig: dexConfig{
-			baseUrl:           fmt.Sprintf("https://dex.%s", domain),
-			authorizeEndpoint: fmt.Sprintf("https://dex.%s/auth", domain),
-			tokenEndpoint:     fmt.Sprintf("https://dex.%s/token", domain),
+			baseUrl:           fmt.Sprintf("https://dex.%s", cfg.DomainName),
+			authorizeEndpoint: fmt.Sprintf("https://dex.%s/auth", cfg.DomainName),
+			tokenEndpoint:     fmt.Sprintf("https://dex.%s/token", cfg.DomainName),
 		},
+
 		clientConfig: clientConfig{
 			id:          clientId,
 			redirectUri: "http://127.0.0.1:5555/callback",
 		},
 
 		userCredentials: userCredentials{
-			username: username,
-			password: password,
+			username: cfg.DexUserEmail,
+			password: cfg.DexUserPassword,
 		},
 	}
 
-	idTokenProvider := newDexIdTokenProvider(dexHttpClient, idProviderConfig)
+	idTokenProvider := newDexIdTokenProvider(client, idProviderConfig)
 
 	Convey("Should issue an ID token", t, func() {
```

Comment:
in line 85 why u didnt just use the `So(err, ShouldBeNil)`?

after 
`tokenParts := strings.Split(idToken, ".")`

you should check that `tokenParts` has min 2 items
`So(tokenParts, ShouldHaveLength, 2)`

cause this statement in line 79 `tokenPayloadEncoded := tokenParts[1]` can panic